x/incentives/keeper/distribute.go

```diff
@@ -266,6 +266,71 @@ func (k Keeper) distributeSyntheticInternal(
 	return k.distributeInternal(ctx, gauge, sortedAndTrimmedQualifiedLocks, distrInfo)
 }
 
+// AllocateAcrossGauges gets all the active groupGauges and distributes tokens evenly based on the internalGauges set for that
+// groupGauge. After each iteration we update the groupGauge by modifying filledEpoch and distributed coins.
+// TODO: Replace even split by volume split once its implemented.
+func (k Keeper) AllocateAcrossGauges(ctx sdk.Context) error {
+	currTime := ctx.BlockTime()
+
+	groupGauges, err := k.GetAllGroupGauges(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, groupGauge := range groupGauges {
+		gauge, err := k.GetGaugeByID(ctx, groupGauge.GroupGaugeId)
+		if err != nil {
+			return err
+		}
+
+		// only allow distribution if the GroupGauge is Active
+		if !currTime.Before(gauge.StartTime) && (gauge.IsPerpetual || gauge.FilledEpochs < gauge.NumEpochsPaidOver) {
```

Comment:
nit: for general readability, I think doing something like "if currTime.After(startTime) && (!gaugeIsActive bool), then `continue`" would be cleaner instead of having half the function in an if branch.

Ofc feel free to ignore this if you think the current approach is more readable (hence the nit) – my thinking is just that you need to keep track of fewer things mentally when you're not in a branch

Reply:
yup i think we can just replace this with 
`gauge.IsActiveGauge(currTime)` as well. Looks cleaner